### PR TITLE
Check for invalid L7-ILB HTTPS configuration

### PR DESCRIPTION
**Pull request by spencerhance**
Create a WillNotConfigureFrontend event warning the user if they set up an internal https ingress without the allow-http annotation set to false, since that functionality is currently not supported.

### CODE REVIEW

File: pkg/loadbalancers/l7.go [hunk 1]
```diff
@@ -155,6 +156,16 @@ func (l *L7) edgeHop() error {
 	// Keeps track if we will "try" to setup frontend resources based on user configuration.
 	// If user configuration dictates we do not, then we emit an event.
 	willConfigureFrontend := false
+	sslConfigured := l.runtimeInfo.TLS != nil || l.runtimeInfo.TLSName != ""
+
+	// Check for invalid L7-ILB HTTPS config before attempting sync
+	if flags.F.EnableL7Ilb &&
```
Comment (bowei) [inline pkg/loadbalancers/l7.go]: put this all on a single line (no line limits for golang style)

**Comment by bowei** [comment]
/ok-to-test

Shouldn't there be a unit test for this?

File: pkg/loadbalancers/l7.go (hunk 1 above)
Comment (spencerhance) [reply pkg/loadbalancers/l7.go]: Done

**Comment by spencerhance** [comment]
@bowei added a unit test as well

**Comment by spencerhance** [comment]
@bowei following up on this

**Comment by spencerhance** [comment]
Fixed the prow test

File: pkg/loadbalancers/loadbalancers_test.go [hunk 2]
```diff
@@ -207,6 +207,27 @@ func TestCreateHTTPSILBLoadBalancer(t *testing.T) {
...
+	gceUrlMap.DefaultBackend = &utils.ServicePort{NodePort: 31234, BackendNamer: j.namer}
+	gceUrlMap.PutPathRulesForHost("bar.example.com", []utils.PathRule{{Path: "/bar", Backend: utils.ServicePort{NodePort: 30000, BackendNamer: j.namer}}})
+	lbInfo := &L7RuntimeInfo{
+		AllowHTTP: true,
+		TLS:       []*TLSCerts{createCert("key", "cert", "name")},
+		UrlMap:    gceUrlMap,
+		Ingress:   newILBIngress(),
+	}
+
+	_, err := j.pool.Ensure(lbInfo)
```
Comment (bowei) [inline pkg/loadbalancers/loadbalancers_test.go]: if _, err := ... ; err == nil {
  t.Fatalf("j.pool.Ensure(%v) = nil, want err"...)
}

Comment (spencerhance) [reply pkg/loadbalancers/loadbalancers_test.go]: done

**Pull request merged**